app/server/lib/oidc/Protections.ts

```diff
@@ -0,0 +1,123 @@
+import { StringUnion } from 'app/common/StringUnion';
+import { SessionObj } from 'app/server/lib/BrowserSession';
+import { AuthorizationParameters, generators, OpenIDCallbackChecks } from 'openid-client';
+
+export const EnabledProtection = StringUnion(
+  "STATE",
+  "NONCE",
+  "PKCE",
+);
+export type EnabledProtectionString = typeof EnabledProtection.type;
+
+type SessionOIDCInfo = Exclude<SessionObj['oidc'], undefined>;
+
+interface Protection {
```

Comment:
This file is beautiful! I like how you cleanly you organized it, that it's easy to read and understand even without comments.

Reply:
Thanks to your suggestion!